packages/ember-uploader/lib/file-field.js

```diff
@@ -1,13 +1,22 @@
+var deprecate = Ember.deprecate;
 var set = Ember.set;
+var on = Ember.on;
 
-export default Ember.TextField.extend({
+export default Ember.TextField.extend(Ember.Evented, {
   type: 'file',
   attributeBindings: ['multiple'],
   multiple: false,
   change: function(e) {
     var input = e.target;
     if (!Ember.isEmpty(input.files)) {
+      this.trigger('filesDidChange', input.files);
       set(this, 'files', input.files);
```

Comment:
I left it in there so we can issue a deprecation warning to people using the observer and then we can remove that in a future release. If you'd prefer, I can remove it completely, but I think it'll be more in line with the ember way if we deprecate it first